src/datetime.rs

```diff
@@ -206,32 +206,25 @@ impl<U: Unit> fmt::Debug for Timedelta<U> {
 
 struct TypeDescriptors {
     npy_type: NPY_TYPES,
-    dtypes: UnsafeCell<Option<FxHashMap<NPY_DATETIMEUNIT, Py<PyArrayDescr>>>>,
+    #[allow(clippy::type_complexity)]
+    dtypes: GILProtected<RefCell<Option<FxHashMap<NPY_DATETIMEUNIT, Py<PyArrayDescr>>>>>,
```

Comment:
A thought inspired by the discussion on enums in PyO3 - `NPY_DATETIMEUNIT` is a dense enum, so this can potentially be changed from an `FxHashMap` to an array. Probably for a future PR.

Reply:
> A thought inspired by the discussion on enums in PyO3 - NPY_DATETIMEUNIT is a dense enum, so this can potentially be changed from an FxHashMap to an array. Probably for a future PR.

I actually implemented this but IIRC the initialization would have to open-code the correct number of `None`s or use some macro trickery because `Option<Py<PyArrayDescr>>` is not `Copy` so that `[None; NPY_DATETIMEUNIT::NPY_FR_GENERIC as usize]` does not work and `[(); NPY_DATETIMEUNIT::NPY_FR_GENERIC as usize].map(|()| None)` is not yet within MSRV reach (needs 1.55, so would be available after the Debian Bookworm bump).